ApiFirst/ApiFirst/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,102 @@
+﻿using AppModels.AppModels.Products;
+using AppServices.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace ApiFirst.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductsController : ControllerBase
+    {
+        private readonly IProductAppService _productAppService;
+
+        public ProductsController(IProductAppService repository)
+        {
+            _productAppService = repository ?? throw new ArgumentNullException(nameof(repository));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllAsync()
+        {
+            try
+            {
+                var portolios = await _productAppService.GetAllAsync();
+                return Ok(portolios);
+            }
```

Comment:
Faça isso em todas as outras funções async

```suggestion
            {
                var portolios = await _productAppService.GetAllAsync().ConfigureAwait(false);
                return Ok(portolios);
            }
```